packages/@aws-cdk/aws-ecs-patterns/lib/base/scheduled-task-base.ts

```diff
@@ -140,6 +148,11 @@ export abstract class ScheduledTaskBase extends CoreConstruct {
    */
   public readonly subnetSelection: SubnetSelection;
 
+  /**
+   * The IAM Role to use for CloudWatch Event invocation.
+   */
+  public readonly eventRole?: IRole;
```

Comment:
This public property is not very useful, since it only surfaces the eventRole if the user passed in their own role through props. If they passed in the role through props, then they already have a reference to it somewhere in their CDK code. And if they did not provide a role, then this will be undefined and users will be frustrated that it doesn't reference the role created by CDK. 

So I think we should remove this or make it surface the role created in the `EcsTask` by default.

Reply:
making the whole l3 construct able to be referenced via `grant` would make more sense imho